src/base/SubtreeTile.js

```diff
@@ -0,0 +1,45 @@
+export class SubtreeTile {
+
+
+	constructor( parentTile, childMortonIndex ) {
+
+		this.parent = parentTile;
+
+
+		// Number of leaves in the subtree	(should go from 0 to 4 (Quadtree) or 8 (Octree)
+		this.__depth = parentTile.__depth + 1;
+
+		//index inside of the tree
+		this.__subtreeIdx = childMortonIndex;
+
+
+		this.__basePath = parentTile.__basePath;
+
+		let coord = this.getSubtreeCoordinates(this, parentTile);
+		this.__x = coord.x;
+		this.__y = coord.y;
+
+
+
+	}
+
+
+	getSubtreeCoordinates( tile, parentTile ) {
+
+		if (!parentTile) {
+			return {x: 0, y: 0}
+		}
+
+		const x = 2 * parentTile.__x + (tile.__subtreeIdx % 2);
+		const y = 2 * parentTile.__y + (Math.floor(tile.__subtreeIdx / 2) % 2);
+
+
+		return {x, y};
+
+	};
```

Comment:
Is there a reason we need a special class with a unique member function rather than a assigning these onto the tile object?

Reply:
It was to assign some attributes. Kind of overkill, I agree
Would you rather have a function in the subtreeloader adding these attributes ?

Reply:
I think it's okay to just create a bare javascript object with these fields pre-computed - or ideally compute them as-needed in the associated functions if it doesn't result in too much redundancy. I think ideally we'd be returning a tile hierarchy that shaped as much like the native json [tile schema](https://github.com/CesiumGS/3d-tiles/blob/main/specification/schema/tile.schema.json) from the loader as possible. Then we can rely on the rest of the tiles rendering to just work.